packages/starknet-snap/src/chain/data-client/starkscan.ts

```diff
@@ -0,0 +1,280 @@
+      deployTx && txs.push(deployTx);
+    }
+
+    return txs;
+  }
+
+  /**
+   * Fetches the deploy transaction for a given contract address.
+   *
+   * @param address - The address of the contract to fetch the deploy transaction for.
+   * @returns A Promise that resolve the Transaction object or null if the transaction can not be found.
+   */
+  async getDeployTransaction(address: string): Promise<Transaction | null> {
+    // Fetch the first 5 transactions to find the deploy transaction
+    // The deploy transaction usually is the first transaction from the list
+    const apiUrl = this.getApiUrl(
+      `/transactions?contract_address=${address}&order_by=asc&limit=5`,
+    );
+
+    const result = await this.sendApiRequest<StarkScanTransactionsResponse>({
+      apiUrl,
+      responseStruct: StarkScanTransactionsResponseStruct,
+      requestName: 'getTransactions',
+    });
+
+    for (const data of result.data) {
+      if (this.isDeployTransaction(data)) {
+        return this.toTransaction(data);
+      }
+    }
+
+    return null;
+  }
+
+  protected isDeployTransaction(tx: StarkScanTransaction): boolean {
+    return tx.transaction_type === TransactionType.DEPLOY_ACCOUNT;
+  }
+
+  protected isFundTransferTransaction(entrypoint: string): boolean {
+    return entrypoint === 'transfer';
+  }
+
+  protected getContractAddress(tx: StarkScanTransaction): string {
+    // backfill the contract address if it is null
+    return tx.contract_address ?? '';
+  }
+
+  protected getSenderAddress(tx: StarkScanTransaction): string {
+    let sender = tx.sender_address;
+
+    if (this.isDeployTransaction(tx)) {
+      // if it is a deploy transaction, the contract address is the sender address
+      sender = tx.contract_address as unknown as string;
+    }
+
+    // backfill the sender address if it is null
+    return sender ?? '';
+  }
+
+  protected toTransaction(tx: StarkScanTransaction): Transaction {
+    /* eslint-disable @typescript-eslint/naming-convention */
+    const {
+      transaction_hash: txnHash,
+      transaction_type: txnType,
+      timestamp,
+      transaction_finality_status: finalityStatus,
+      transaction_execution_status: executionStatus,
+      max_fee: maxFee,
+      actual_fee: actualFee,
+      revert_error: failureReason,
+      account_calls: calls,
+    } = tx;
+
+    // account_calls representing the calls to invoke from the account contract, it can be multiple
+    // If the transaction is a deploy transaction, the account_calls is a empty array
+    const accountCalls = this.toAccountCall(calls);
+
+    return {
+      txnHash,
+      txnType,
+      chainId: this.network.chainId,
+      senderAddress: this.getSenderAddress(tx),
+      timestamp,
+      finalityStatus,
+      executionStatus,
+      maxFee,
+      actualFee,
+      contractAddress: this.getContractAddress(tx),
+      accountCalls,
+      failureReason: failureReason ?? '',
+      version: 'V2',
```

Comment:
Why don't we include version from the response here ?

Reply:
yea you are right, may be it is better to include a version of the transaction